data/helpers.d/package

```diff
@@ -55,48 +71,67 @@ ynh_package_install() {
 #
 # usage: ynh_package_install_from_equivs controlfile
 # | arg: controlfile - path of the equivs control file
-ynh_package_install_from_equivs() {
+ynh_package_install_from_equivs () {
     controlfile=$1
 
-    # install equivs package as needed
+    # Check if the equivs package is installed. Or install it.
     ynh_package_is_installed 'equivs' \
         || ynh_package_install equivs
 
     # retrieve package information
-    pkgname=$(grep '^Package: ' $controlfile | cut -d' ' -f 2)
-    pkgversion=$(grep '^Version: ' $controlfile | cut -d' ' -f 2)
+    pkgname=$(grep '^Package: ' $controlfile | cut -d' ' -f 2)	# Retrieve the name of the debian package
+    pkgversion=$(grep '^Version: ' $controlfile | cut -d' ' -f 2)	# And its version number
     [[ -z "$pkgname" || -z "$pkgversion" ]] \
-        && echo "Invalid control file" && exit 1
+        && echo "Invalid control file" && exit 1	# Check if this 2 variables aren't empty.
 
-    # update packages cache
+    # Update packages cache
     ynh_package_update
 
-    # build and install the package
+    # Build and install the package
     TMPDIR=$(ynh_mkdir_tmp)
-    (cp "$controlfile" "${TMPDIR}/control" \
-     && cd "$TMPDIR" \
-     && equivs-build ./control 1>/dev/null \
+    cp "$controlfile" "${TMPDIR}/control" && pushd "$TMPDIR"	# pushd is like a cd, but it stores the previous directory
+    # Create a fake deb package with equivs-build and the given control file
+    # Install the fake package without its dependencies with dpkg
+    # Install missing dependencies with ynh_package_install
+    equivs-build ./control 1>/dev/null \
      && sudo dpkg --force-depends \
           -i "./${pkgname}_${pkgversion}_all.deb" 2>&1 \
-     && ynh_package_install -f) \
-        && ([[ -n "$TMPDIR" ]] && rm -rf $TMPDIR)
+     && ynh_package_install -f
+    [[ -n "$TMPDIR" ]] && rm -rf $TMPDIR	# Remove the temp dir.
 
     # check if the package is actually installed
     ynh_package_is_installed "$pkgname"
+    popd	# Like a cd on the directory stored by pushd
 }
 
-# Remove package(s)
+# Install dependencies with a equivs control file
 #
-# usage: ynh_package_remove name [name [...]]
-# | arg: name - the package name to remove
-ynh_package_remove() {
-    ynh_apt remove $@
+# usage: ynh_app_dependencies dep [dep [...]]
+# | arg: dep - the package name to install in dependence
```

Comment:
We talked about this on the chat quickly with @maniackcrudelis : 
- the main issue is that there are sometimes optionnal dependencies. @maniackcrudelis cited the case of Abiword for etherpad (Abiword is optionnal and weights 300 Mo)
- therefore simply having a "requirements" field in the app manifest is not flexible enough
- however, having these in the app manifest had the advantage of having a clean and simple way to know what an app depends on for usages outside the app's scripts (e.g. from the yunohost core, for some use case we don't know yet)
- the solution is to add a new app setting with the list of dependencies *effectively used*. So for instance, if etherpad installs abiword, it should be listed in that setting. If not, then it's not listed. Simply calling `ynh_app_dependencies` should append the new dependencies to this setting.

By the way, here we're talking about debian packages, but the dependency/requirement problematic is much larger if we start talking about apps which include sources (e.g. .tar.gz) of fetch sources (wget, git clone, ...). For example with Wekan I had to install meteor, which can be considered as an "intermediate" dependency, with a wget. What if another app needs meteor ? It's pretty complicated to keep track and handle these dependencies properly :/ ...